Volume/View Models/MagazinesListViewModel.swift

```diff
@@ -2,121 +2,220 @@
+                    for: GetMagazinesBySemesterQuery(
+                        semester: selectedSemester,
+                        limit: Constants.moreMagazinesLimit,
+                        offset: offset(for: moreMagazines)
+                    )
+                )
+                .map { $0.magazines.map(\.fragments.magazineFields) }
+                .sink { [weak self] completion in
+                    self?.networkState?.handleCompletion(screen: .magazines, completion)
+                } receiveValue: { [weak self] magazineFields in
+                    Task {
+                        let newMagazines = await [Magazine](magazineFields)
+                        self?.loadMoreMagazines(with: newMagazines)
                     }
                 }
-            }
+                .store(in: &queryBag)
         }
         
-        func fetchAllMagazines() {
-            sectionStates.magazinesBySemester = .loading
-            sectionQueries.magazinesBySemester = Network.shared.publisher(
-                for: GetAllMagazinesQuery(limit: Constants.allMagazinesLimit, offset: offset)
-            )
-            .map { $0.magazines.map(\.fragments.magazineFields) }
-            .sink { [weak self] completion in
-                self?.networkState.handleCompletion(screen: .magazines, completion)
-            } receiveValue: { [weak self] magazineFields in
-                Task {
-                    let allMagazines = await [Magazine](magazineFields)
-                    withAnimation(.linear(duration: 0.1)) {
-                        self?.sectionStates.magazinesBySemester = .results(allMagazines)
+        private func fetchAllSemestersNextPage() {
+            Network.shared
+                .publisher(
+                    for: GetAllMagazinesQuery(
+                        limit: Constants.moreMagazinesLimit,
+                        offset: offset(for: moreMagazines)
+                    )
+                )
+                .map { $0.magazines.map(\.fragments.magazineFields) }
+                .sink { [weak self] completion in
+                    self?.networkState?.handleCompletion(screen: .magazines, completion)
+                } receiveValue: { [weak self] magazineFields in
+                    Task {
+                        let newMagazines = await [Magazine](magazineFields)
+                        self?.loadMoreMagazines(with: newMagazines)
                     }
                 }
-            }
+                .store(in: &queryBag)
         }
         
-        // MARK: Helpers
+        // MARK: - Helpers
+        
+        private func loadMoreMagazines(with newMagazines: [Magazine]) {
+            switch moreMagazines {
+            case .none:
+                moreMagazines = newMagazines
+            case .some(let oldMagazines):
+                moreMagazines = oldMagazines + newMagazines
+            }
 
+            if newMagazines.count < Int(Constants.moreMagazinesLimit) {
+                hasMoreMagazines = false
```

Comment:
why is this set to false when number of new magazines is less than limit

Reply:
We want to fetch 4 magazines every time. If we ended up only fetching 3, this means that there are no more magazines to be fetched.